### ISSUE TITLE
hdf files not being saved into database on cleanup

### ISSUE THREAD

**Issue by leoscholl** [aolabNeuro/brain-python-interface#27]
no HDF file was saved to the database nor copied from /tmp/ into the HDF directory
web UI never exited the "running" state, despite pressing the "Stop Experiment" button

*************************** STARTING TASK *****************************
sinkmanager start <class 'hdfwriter.hdfwriter.HDFWriter'>
HDFWriter: Saving datafile to /tmp/tmp49lxen29.h5
HDFWriter: opening file
HDFWriter registered 'optitrack'
HDFWriter registered 'task'
Compiling shader passthru...
Compiling shader default...
runseq
end of FSM.run, task state is None
Stopping optitrack
Running task cleanup functions....
Closed hdf
ended datasink

**Comment by leoscholl** [comment]
no errors when this happens in any of the logs, just missing the final cleanup steps that normally happen:

```
Tasktrack is stopping the task...
[17/Feb/2021 22:56:31] "POST /exp_log/stop/ HTTP/1.1" 200 58
Running task cleanup functions....

Joining optitrack datasource
Calling saveout/task cleanup code
experiment.LogExperiment.cleanup
experimient.Experiment.cleanup executing
save_log {'runtime': 4.921511173248291, 'n_trials': 2, 'n_success_trials': 0}
WebSocket closed
Beginning HDF file cleanup
	HDF data currently saved to temp file: /tmp/tmpk_p8xfcu.h5
	Running self.cleanup_hdf()
[17/Feb/2021 22:56:31] "POST /RPC2/ HTTP/1.1" 200 114
copying file...
Saved datafile for file=/tmp/tmpk_p8xfcu.h5 -> test20210217_29_te816.hdf, system=hdf, id=816)...
[17/Feb/2021 22:56:31] "POST /RPC2/ HTTP/1.1" 200 114
Saving optitrack file to database...
Saved datafile for file=C:/Users/Orsborn Lab/Documents/OptiTrack/Session 2021-02-17/Take 2021-02-17 22:56:25 -> C:/Users/Orsborn Lab/Documents/OptiTrack/Session 2021-02-17/Take 2021-02-17 22:56:25, system=optitrack, id=816)...
[17/Feb/2021 22:56:31] "POST /RPC2/ HTTP/1.1" 200 114
...done.
*************************** EXITING TASK *****************************
```

compared to:

```
Tasktrack is stopping the task...
[17/Feb/2021 22:56:21] "POST /exp_log/stop/ HTTP/1.1" 200 58
Running task cleanup functions....

Joining optitrack datasource
```

or sometimes:

```
Tasktrack is stopping the task...
[17/Feb/2021 22:52:22] "POST /exp_log/stop/ HTTP/1.1" 200 58
Running task cleanup functions....

Joining optitrack datasource
Calling saveout/task cleanup code
*************************** EXITING TASK *****************************
```

**Comment by leoscholl** [comment]
fixed by 9a43dba816e1e0430ee1da9420107bf643e9677a. simply needed to stop the optitrack datasource in the None state
